marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mapping/getDocument.sjs

```diff
@@ -14,13 +14,42 @@
 xdmp.securityAssert("http://marklogic.com/data-hub/privileges/read-mapping", "execute");
 
 const core = require('/data-hub/5/artifacts/core.sjs')
+const Artifacts = require('/data-hub/5/artifacts/core.sjs');
+const flowRunner = require("/data-hub/5/flow/flowRunner.sjs");
+const httpUtils = require("/data-hub/5/impl/http-utils.sjs");
 
 var stepName, uri;
-let mappingStep = core.getArtifact("mapping", stepName);
+
+const mappingStep = core.getArtifact("mapping", stepName);
+
+let stepExecutionContext = {};
+stepExecutionContext.flowStep = mappingStep;
+console.log(mappingStep.stepId);
+stepExecutionContext.combinedOptions = Artifacts.convertStepReferenceToInlineStep(mappingStep.stepId, null).options;
+
+let originalSourceDoc;
 if(mappingStep.sourceDatabase) {
-  fn.head(xdmp.eval("String(cts.doc('" + uri + "'))", null, {database: xdmp.database(mappingStep.sourceDatabase)}));
+  originalSourceDoc = fn.head(xdmp.eval("cts.doc('" + uri + "')", null, {database: xdmp.database(mappingStep.sourceDatabase)}));
 }
 else{
-  String(cts.doc(uri));
+  originalSourceDoc = cts.doc(uri);
+}
+
+const contentArray = [];
+let content = {};
+content.uri = uri;
+content.value = originalSourceDoc;
+contentArray.push(content);
+
+try{
+  flowRunner.invokeInterceptors(stepExecutionContext, contentArray, "beforeMain");
 }
+catch(e){
+  httpUtils.throwBadRequest("Interceptor execution failed;cause: " + e.message);
+}
+
+JSON.stringify(contentArray[0].value);
```

Comment:
Does this work for XML too? I checked and I don't think there are any tests on this endpoint for an XML source document. I think this PR should fill that gap, should just be a new test in MappingStepControllerTest.